Update ece-architecture.md - Adding a container <-> ECE host role mapping table (#1369)
elastic/docs-content: deploy-manage/deploy/cloud-enterprise/ece-architecture.md

    Docker containers communicate securely with one another through Transport Layer Security, provided by [Stunnel](https://www.stunnel.org/) (as not all of the services or components support TLS natively). Tunneling all traffic between containers makes sure that it is not possible to eavesdrop, even when someone else has access to the underlying cloud or network infrastructure.

## ECE service containers per host role(s) [ece-service-containers]

> Each Elastic Cloud Enterprise service runs as a dedicated Docker container. These containers are automatically deployed based on the roles assigned to each ECE host.

Reviewer: ```suggestion
Each Elastic Cloud Enterprise service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host.
```

What about removing the word Docker considering we also support Podman?

Reviewer 2: based on this note, we might need to remove it from the section above as well ("services as docker containers")

Author: Thanks @eedugon and @shainaraskas.  I'm all for omitting the docker references (it's mentioned 6 times on the page) and improving the entire page as a whole, but it is out of scope of my proposal of documenting the service containers.

Reviewer: ^^ don't worry about that @rseldner , we could remove `Docker` references if needed afterwards. I just wanted to ensure we didn't add more in the PR :D

Author: agreed. thanks!

| Container                                          | Roles         | Description |
|---|---|---|
| `frc-admin-consoles-admin-console`                  | Controller    | Backend service for the ECE UI; handles API requests and coordinates with ZooKeeper, Elasticsearch, logging, and security services. |
| `frc-allocator-metricbeats-allocator-metricbeat`    | Allocator     | Collects allocator metrics via Beats. |
| `frc-allocators-allocator`                          | Allocator     | Manages container lifecycle for Elasticsearch and Kibana; reports host capacity to ZooKeeper. |
| `frc-beats-runners-beats-runner`                    | All roles     | Collects metrics from local containers for monitoring and health checks. |
| `frc-blueprints-blueprint`                          | Director      | Coordinates container startup by providing configuration data to runners based on their role and token. |

> | `frc-client-forwarders-client-forwarder`            | All roles     | Manages communication between nodes and ZooKeeper. |

Reviewer 3: ```suggestion
| `frc-client-forwarders-client-forwarder`            | All roles     | Manages communication between hosts and ZooKeeper. |
```
I think `node` is already used for too many different things and thus suggesting host. Another option could be Runner, but host is more generic and doesn't require ECE knowledge

Author: Agreed with the term `host` and what I would usually use when speaking ECE.

| `frc-cloud-uis-cloud-ui`                            | Controller    | Web frontend for the ECE UI, served to users in the browser. |

> | `frc-constructors-constructor`                      | Controller    | Schedules changes via ZooKeeper; assigns clusters to allocators and balances zones. |

Reviewer: ```suggestion
| `frc-constructors-constructor`                      | Controller    | Schedules and coordinates deployment changes through ZooKeeper; assigns instances to allocators and balances zones. |
```

| `frc-container-task-services-container-task-service`| Allocator     | Supports autoscaling and tracks feature usage. |
| `frc-directors-director`                            | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
| `frc-proxies-proxyv2`                               | Proxy         | Routes user traffic; uses ZooKeeper to map deployment IDs to cluster nodes. |
| `frc-proxies-route-server`                          | Proxy         | Manages internal routing and port mapping for cluster access. |

> | `frc-runners-runner`                                | All roles     | Deploys containers locally per host role definitions. |

Reviewer 3: ```suggestion
| `frc-runners-runner`                                | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |
```

    Docker containers communicate securely with one another through Transport Layer Security, provided by [Stunnel](https://www.stunnel.org/) (as not all of the services or components support TLS natively). Tunneling all traffic between containers makes sure that it is not possible to eavesdrop, even when someone else has access to the underlying cloud or network infrastructure.

## ECE service containers per host role(s) [ece-service-containers]

> Each Elastic Cloud Enterprise service runs as a dedicated Docker container. These containers are automatically deployed based on the roles assigned to each ECE host.

Reviewer: ```suggestion
Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers you’ll find on your ECE hosts:
```

> Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers on ECE hosts, along with the host roles that include each container.:

Reviewer: ```suggestion
Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers on ECE hosts, along with the host roles that include each container:
```

> | Container                                          | Host roles         | Description |

Author: updated with `The following table lists the containers on ECE hosts, along with the host roles that include each container.`

|---|---|---|

> | `frc-beats-runners-beats-runner`                  | All roles     | Collects metrics from local containers for monitoring and health checks. |

Reviewer: done

| `frc-client-forwarders-client-forwarder`          | All roles     | Manages communication between hosts and ZooKeeper. |

> | `frc-runners-runner`                              | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |

Reviewer 4: `according to their defined roles` - this suggests the _containers_ have roles defined, whereas it's _hosts_ the roles pertain to. 
In other words, a host has a role assigned to it and that role defines what containers should run on the host. 

The runner does indeed manage these containers, makes sure they're started at the proper version, etc.  , but saying that it `ensures they are online and healthy` isn't very accurate. I'd rephrase this whole line to sth like:
```
Runs on every ECE host and provides a supervisor service to deploy and manage containers based on the host's role. Ensures the required containers are started at the proper version.
```

Reviewer: done

| `frc-services-forwarders-services-forwarder`      | All roles     | Routes internal service data across the ECE platform. |

> | `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics via Beats. |

Reviewer: ```suggestion
| `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics. |
```

> | `frc-allocators-allocator`                        | Allocator     | Manages container lifecycle for Elasticsearch and Kibana; reports host capacity to ZooKeeper. |

Reviewer: ```suggestion
| `frc-allocators-allocator`                        | Allocator     | Manages container lifecycle for {{es}} and {{kib}} instances; reports host capacity to ZooKeeper. |
```

| `frc-container-task-services-container-task-service` | Allocator  | Supports autoscaling and tracks feature usage. |

> | `frc-admin-consoles-admin-console`                | Controller    | Backend service for the ECE UI; handles API requests and coordinates with ZooKeeper, Elasticsearch, logging, and security services. |

Reviewer: ```suggestion
| `frc-admin-consoles-admin-console`                | Controller    | Backend service for the ECE UI; handles API requests and coordinates with ZooKeeper, {{es}}, logging, and security services. |
```

| `frc-cloud-uis-cloud-ui`                          | Controller    | Web frontend for the ECE UI, served to users in the browser. |
| `frc-constructors-constructor`                    | Controller    | Schedules and coordinates deployment changes through ZooKeeper; assigns instances to allocators and balances zones. |
| `frc-blueprints-blueprint`                        | Controller    | Coordinates container startup by providing configuration data to runners based on their role and token. |

> | `frc-directors-director`                          | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |

Reviewer: done

> | `frc-zookeeper-servers-zookeeper`                 | Director      | Consistent distributed store used to track ECE state and coordinate communication between services. |

Reviewer 4: `store` --> `data store` ?

> | `frc-proxies-proxyv2`                             | Proxy         | Routes user traffic; uses ZooKeeper to map deployment IDs to cluster nodes. |

Reviewer 4: I'd say `Routes user traffic to clusters (Elasticsearch, Kibana, etc.)`. 
The part about using ZooKeeper is also a bit of an implementation detail I think.

Reviewer: Done, but used {{stack}} deployments instead of clusters. Feel free to comment on that if you think we should change it.

> | `frc-proxies-route-server`                        | Proxy         | Manages internal routing and port mapping for cluster access. |

Reviewer 4: `Manages internal routing` may not be very accurate I think? We could say `Manages routing tables used by proxyv2 to route traffic to clusters`.

Reviewer: Made it a bit generic:
> Manages the routing tables used by the proxy service.

Feel free to comment on that
